gpkitmodels/SP/aircraft/prop/propeller.py

```diff
@@ -0,0 +1,127 @@
+    vt                      [m/s]       Tangential induced velocity
+    G                       [m^2/s]     Circulation
+    cl                     [-]          local lift coefficient
+    cd                      [-]         local drag coefficient
+    B           2           [-]         number of blades
+    r                       [m]         local radius
+    lam_w                   [-]         advance ratio
+    eps                     [-]         blade efficiency
+    AR_b                   [-]          blade aspect ratio
+    AR_b_max    50          [-]         max blade aspect ratio
+    Re                      [-]         blade reynolds number
+    f                       [-]         intermediate tip loss variable
+    F                       [-]         Prandtl tip loss factor
+    cl_max      .6         [-]          max airfoil cl
+    dr                      [m]         length of blade element
+    M                       [-]         Mach number
+    a           295         [m/s]       Speed of sound at altitude
+
+    """
+
+    def setup(self,static,  state):
+        exec parse_variables(Blade_Element_Performance.__doc__)
+
+        V       = state.V
+        rho     = state.rho
+        R       = static.R
+        mu      = state.mu
+        path = os.path.dirname(__file__)
+        fd = pd.read_csv(path + os.sep + "dae51_fitdata.csv").to_dict(
+            orient="records")[0]
+        c = static.c
+        constraints = [TCS([Wa>=V + va]),
+                        TCS([Wt + vt<=omega*r]),
+                        TCS([G == (1./2.)*Wr*c*cl]),
+                        F == (2./pi)*(1.01116*f**.0379556)**(1./.1), #This is the GP fit of arccos(exp(-f))
+                        M == Wr/a,
+                        lam_w == (r/R)*(Wa/Wt),
+                        va == vt*(Wt/Wa),
+                        eps == cd/cl,
+                        TCS([dQ >= rho*B*G*(Wa+eps*Wt)*r*dr]),
+                        AR_b == R/c,
+                        AR_b <= AR_b_max,
+                        Re == Wr*c*rho/mu,
+                        eta_i == (V/(omega*r))*(Wt/Wa),
+                        TCS([f+(r/R)*B/(2*lam_w) <= (B/2.)*(1./lam_w)]),
+                        
+                        XfoilFit(fd, cd, [cl,Re], name="polar"),
+                        
+                        cl <= cl_max
+                    ]
+        with SignomialsEnabled():
+            constraints += [SignomialEquality(Wr**2,(Wa**2+Wt**2)),
+                            
+                            TCS([dT <= rho*B*G*(Wt-eps*Wa)*dr]),
+                            #SignomialEquality(vt**2*F**2*(1.+(4.*lam_w*R/(pi*B*r))**2),(B*G/(4.*pi*r))**2),
+                            TCS([vt**2*F**2*(1.+(4.*lam_w*R/(pi*B*r))**2)>=(B*G/(4.*pi*r))**2]),
+                            #SignomialEquality(f+(r/R)*B/(2*lam_w),(B/2.)*(1./lam_w))
+                            
+                    
+            ]
+        return constraints, state
+
+
+
+class MultiElementProp(Model):
```

Comment:
I think this should be PropPerf or QPropPerf.  Multielement is maybe a little ambiguous